### ISSUE TITLE
"Super expression must either be null or a function" error in React

### ISSUE THREAD

**Issue by trebor** [observablehq/plot#619]
when i build and run a deployment version of a [CRA](https://create-react-app.dev/) based app which uses `Plot` i get the following error.

![image](https://user-images.githubusercontent.com/12802394/146613245-5072fd2b-0c0f-4028-a96d-827417a8cec6.png)

i get the error before anything runs, long before the `Plot` code is called, so i think it has to do with the package construction.

it runs fine in dev mode `npm run start`.  it only fails in the built version.

[This artical](https://medium.com/@anuhosad/react-errors-super-expression-must-either-be-null-or-a-function-4aac6e5173ad)  mentions one possible cause is circular dependencies:

> **Circular Dependencies**
> This is a an often missed culprit which I came across recently in one of the projects I was working on. After going into the depths of the google search rabbit hole, I finally came across a stack overflow answer that shed some light on why I was still getting the error in spite of the component being exported / imported correctly.
> If you have dependencies of a cyclic nature like shown below, it will result in this error:
>
> ![image](https://user-images.githubusercontent.com/12802394/146613739-a4e5a52e-6b42-4886-98d3-324d0271fbe9.png)

when i build @observable/plot locally:
~~~
$ npm run prepublishOnly
~~~

i see:

![image](https://user-images.githubusercontent.com/12802394/146613813-851d7992-c25a-45a1-aa6a-5aac640f1641.png)

these are the what i assume to be the relevant deps in my `package.json`:

~~~
    "@observablehq/plot": "^0.3.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router": "^6.1.1",
    "react-router-dom": "^6.1.1",
    "react-scripts": "^4.0.3",
~~~

and my env:

~~~ 
     node --version  v14.16.1
     npm --version  6.14.14
~~~

happy to proved more deets, but the circular dependencies seems like it might be worth getting worked out anyway.

**Comment by mbostock** [comment]
Circular dependencies are valid in ES imports:

https://exploringjs.com/es6/ch_modules.html#sec_rationale-cyclic-dependencies

The Rollup warnings are meaningless here; they’re not something we need to fix. (We may end up configuring Rollup to suppress these warnings because people often file issues complaining that this is an error when it is not.)

But, I’m not sure why create-react-app/Babel is having trouble here. I’m generally loathe to workaround tools that don’t work correctly with standard JavaScript, but if the workaround is easy, I might consider it.

**Comment by trebor** [comment]
thanks for the quick response!

fyi i did find another [reference the same problem](https://github.com/iddan/plot-react/issues/1), but the fix motioned there didn't work for me.

i've tried quite a few things at the point, but no matter what i do, a if call to Plot.**[anything]**() is present in the code, the build fails to load in the browser before any code executes.

i've got a local version of plot to play with, and will keep trying things on my side - though build mojo is not exactly my strong suit. :)

**Comment by mbostock** [comment]
I am not able to reproduce this problem. I’ve created a standalone example of Observable Plot and Create React App here which you can try:

https://github.com/observablehq/plot-create-react-app-example

I’ve also created a live example on CodeSandbox:

https://codesandbox.io/s/plot-create-react-app-example-e6o29

Neither exhibited any issues, so I suspect it might be something else specific to your repository. Let me know if I can assist further.

**Comment by trebor** [comment]
ya, it must be something on my end. thanks so much for digging in deep here!

**Comment by mbostock** [comment]
This is now fixed in main, and shortly Plot 0.4.0. Hope that helps!

**Comment by trebor** [comment]
oh, fantastic! thank you!
